pkg/meta/base.go

```diff
@@ -948,22 +950,47 @@ func (m *baseMeta) nextInode() (Ino, error) {
 	m.freeMu.Lock()
 	defer m.freeMu.Unlock()
 	if m.freeInodes.next >= m.freeInodes.maxid {
-		v, err := m.en.incrCounter("nextInode", inodeBatch)
-		if err != nil {
-			return 0, err
+		m.prefetchMu.Lock() // Wait until prefetchInodes() is done
+		nextLimit := m.prefetchedInode
```

Comment:
use m.prefetchedInode to overwrite m.freeInodes

Reply:
Yes, this is what the following code does.

Reply:
```
if m.freeInodes.next >= m.freeInodes.maxid {
   m.prefetchMu.Lock()
   m.freeInodes = m.prefetchedInodes
   m.prefetchedInodes = freeID{}
   m.prefetchMu.Unlock()
}
if m.freeInodes.next >= m.freeInodes.maxid {
}
```

Reply:
Refactored like this?https://github.com/juicedata/juicefs/pull/5130/commits/6043c1c4049f184c67c202536abe54372c87d951